### ISSUE TITLE
[USD-01C] Variable Mutability Specifier (Immutable)

### ISSUE THREAD

**Issue by vm06007** [wise-foundation/lending-audit#90]
## USD-01C: Variable Mutability Specifier (Immutable)

| Type | Severity | Location |
| ---- | -------- | -------- |
| [Gas Optimization](/appendix/finding-types#gas-optimization) | <img class="o-severity o-informational" src="https://omniscia.io/report-assets/informational.png" /> | [USDEquivalent.sol:L9](https://github.com/wise-foundation/lending-audit/blob/4470403c94e81c7e523364d77122bd30b045ae34/contracts/USDEquivalent.sol#L9) |

### Description:

The linked variable is assigned to only once during the contract's `constructor`.

### Example:

```solidity
address public tokenAddress;

constructor(
    int256 _defaultValue,
    uint8 _decimals,
    address _tokenAddress
) {
    returnValue = _defaultValue;
    dec = _decimals;
    tokenAddress = _tokenAddress;
}
```

### Recommendation:

We advise it to be set as `immutable` greatly optimizing its read-access gas cost.

**Comment by vm06007** [comment]
USDEquivalent.sol is not part of the audit and only used for testing purposes to mimic USD value that team can programmatically set to desired value during test to simulate when certain tokens or assets depeg or go up/down from their value. This contract is not used in production and should not have been included in the audit
